docs/designs/build-page.md

```diff
@@ -0,0 +1,243 @@
+# Build page workflow
+
+> It doesn't include the build workflow for Data Model(like `ContextObject`), docfx treat Data Model as resource, the output of Data Model is the transformed input.
+
+## Input
+
+- markdown file (conceptual)
+  - markdown content
+  - metadata defined in `yml header`
+
+- yml/json file (schema driven)
+  - input model
+  - metadata defined in `metadata` section
+
+- config file
+  - global metadata
+  - file metadata
+
+## Build Metadata
+
+> `Rectangle parts` represent build outputs(metadata)
+
+### Workflow
+  
+  ![build-metadata-workflow](./images/build-pipeline-output-metadata.png)
+
+### Details
+- Get `Input Metadata` from file(`yml header` or `metadata section`) & config(`global/file metadata`)
+
+  *markdown example:*
+  ```md
+  ---
+  ms.author: docfx
+  author: docfx
+  ms.updated_at: "8/13/2019"
+  ---
+
+  # title
+  markdown content
+  ```
+
+  *yml example:*
+  ```yml
+  metadata: 
+    ms.author: docfx
+    author: docfx
+    ms.updated_at: "8/13/2019"
+  xref: uid
+  ```
+
+- Create `System Metadata` (including `document_id`, `git_content_url`...) based on `Input Metadata`
+
+  *system metadata example:*
+  ```json
+  {
+      "document_id": "58f52203-474e-4b6e-bb85-13cedd56575e",
+      "git_content_url": "https://github.com/dotnet/docfx/docs/build-page.md",
+      "locale": "en-us",
+      "canonical_url": "https://docs.microsoft.com/docfx/build-page"
+      ...
+  }
+  ```
+
+- [SDP] Transform `Input Metadata` based on `JSON Schema`
+
+- Merge `System Metadata` into `Input Metadata` to create `Output Metadata`
+ 
+  *system metadata example:*
+  ```json
+  {
+    "ms.author": "docfx",
+    "ms.updated_at": "8/13/2019",
+    "author": "docfx",
+    "document_id": "58f52203-474e-4b6e-bb85-13cedd56575e",
+    "git_content_url": "https://github.com/dotnet/docfx/docs/build-page.md",
+    "locale": "en-us",
+    "canonical_url": "https://docs.microsoft.com/docfx/build-page"
+    ...
+  }
+  ```
+
+## Build Model
+
+> `Rectangle parts` represent build outputs(model)
+
+### Workflow
+
+  ![build-output-model-workflow](./images/build-pipeline-output-model.png)
```

Comment:
consider `Markdown/Yaml/Json files` --> `SourceModel` since the term intermediate is a bit vague

Reply:
consider use different color or (solid line vs dashed line) to distinguish which process is handled by docfx build only and which process is driven by template